src/ShopBundle/objects/TShopBasket/TShopBasketArticleCore.class.php

```diff
@@ -276,6 +294,12 @@ public function ChangeAmount($dNewAmount)
      * Also updates the dPriceTotalAfterDiscount value.
      *
      * @param TdbShopDiscount $oActingShopDiscount
+     *
+     * @psalm-suppress UndefinedThisPropertyAssignment
+     *
+     * @FIXME `oActingShopDiscount` is set dynamically here and does not seem to be used anywhere. Also, `SetActingDiscount` does not seem to be called anywhere.
```

Comment:
We should define instance-property `oActingShopDiscount` explicitly instead of implicitly